test/dlc_tests/ec2/pytorch/training/test_pytorch_training.py

```diff
@@ -125,6 +125,19 @@ def test_pytorch_gloo(pytorch_training, ec2_connection, gpu_only, py3_only, ec2_
     execute_ec2_training_test(ec2_connection, pytorch_training, test_cmd)
 
 
+@pytest.mark.integration("gloo")
+@pytest.mark.model("resnet18")
+@pytest.mark.parametrize("ec2_instance_type", PT_EC2_CPU_INSTANCE_TYPE, indirect=True)
+def test_pytorch_gloo_cpu(pytorch_training, ec2_connection, cpu_only, py3_only, ec2_instance_type):
+    """
+    Tests gloo backend
+    """
+    if get_processor_from_image_uri(pytorch_training):
+        pytest.skip(f"{__name__} is for CPU image only")
```

Comment:
The returned `processor` from the `get_processor_from_image_uri` can be one of `["eia", "neuron", "cpu", "gpu", "hpu"]`.

If the function fails to find a processor, it throws a `RuntimeError`. Writing the skip condition this way would cause this test to be skipped in all executions. You would want to compare the output of the `get_processor_from_image_uri` to `"cpu"`.

You don't need to do this, though, because `cpu_only` in the test function fixtures would cause `pytorch_training` to be parametrized in such a way that only CPU DLC images are assigned to it.

Could you please change this so that the pytest.skip and its associated if-condition are removed?

Reply:
Note: If you unify the `testPyTorchGlooCpu` script with the `testPyTorchGloo` script, then you wouldn't need another test function specifically for CPU DLCs.